src/prototype_rust_sdk/workflow_context.rs

```diff
@@ -218,21 +186,188 @@ impl WfContext {
     }
 }
 
-struct WFCommandFut {
+/// A Future that can be cancelled.
+/// Used in the prototype SDK for cancelling operations like timers and activities.
+pub trait CancellableFuture<T>: Future<Output = T> {
+    /// Cancel this Future
+    fn cancel(&self, cx: &WfContext);
+}
+
+struct WFCommandFut<T>
+where
+    T: From<UnblockEvent>,
+{
+    /// We need a field that references the template type T, this works
+    unused: PhantomData<T>,
```

Comment:
Can start w/ leading underscore - also doc comment not really necessary here, pretty common pattern.

Reply:
Okay, I'll change